packages/dm-core-plugins/src/generic-list/GenericListPlugin.tsx

```diff
@@ -38,6 +45,22 @@ const Table = styled.table`
   bordercollapse: collapse;
 `
 
+function paginatedItems(
+  allItems: { [key: string]: TGenericObject },
+  currentPage: number,
+  itemsPerPage: number
+): {
+  [key: string]: TGenericObject
+} {
+  const firstItem = currentPage * itemsPerPage
+  const lastItem = firstItem + itemsPerPage
+  const itemKeys = Object.keys(allItems).slice(firstItem, lastItem)
+  const newItems = Object.values(allItems).slice(firstItem, lastItem)
+  return Object.fromEntries(
+    itemKeys.map((key: string, index: number) => [key, newItems[index]])
+  )
```

Comment:
You could instead do this, if you like:
```suggestion
  return Object.fromEntries(Object.entries(allItems).slice(firstItem, lastItem)) 
```

Reply:
Much better! Fixed :+1: